backend/src/config.ts

```
@@ -0,0 +1,24 @@
+import {Keypair, PublicKey} from "@solana/web3.js";
+
+const base64ToUint8Array = (privateKey: string | undefined): Uint8Array | undefined => {
+  if (!privateKey) return;
+  return Uint8Array.from(
+    Buffer.from(privateKey, 'base64')
+      .toString('ascii')
+      // Removes all whitespace and newlines
+      .replace(/[\s\n]+/g, '')
+      .split(',')
+      .map((c) => parseInt(c, 10)));
+}
+
+export const NETWORK = new PublicKey(process.env.NETWORK ?? '');
+export const GATEKEEPER = new PublicKey(process.env.GATEKEEPER ?? '');
+export const MINT = new PublicKey(process.env.MINT ?? '');
+export const GATEKEEPER_ATA = new PublicKey(process.env.GATEKEEPER_ATA ?? '');
+export const NETWORK_ATA = new PublicKey(process.env.NETWORK_ATA ?? '');
+export const FUNDER_ATA = new PublicKey(process.env.FUNDER_ATA ?? '');
+export const FUNDER_AUTHORITY = Keypair.fromSecretKey(
+  new Uint8Array(base64ToUint8Array(process.env.FUNDER_AUTHORITY) ?? []));
+export const GATEKEEPER_AUTHORITY = Keypair.fromSecretKey(
+  new Uint8Array(base64ToUint8Array(process.env.GATEKEEPER_AUTHORITY) ?? []));
```

I think it would be better to throw a loud exception informing the developer that a field wasn't provided. What do you think?

---

Yeah happy with that. Currently it still fails since an empty string is not a valid key. But I agree that is ambiguous